sdks/java/core/src/test/java/org/apache/beam/sdk/transforms/DoFnTesterTest.java

```diff
@@ -126,6 +130,15 @@ public void processElementsWithPeeks() throws Exception {
   }
 
   @Test
+  public void processElementAfterFinish() throws Exception {
+    DoFnTester<Long, String> tester = DoFnTester.of(new CounterDoFn());
+    tester.finishBundle();
+
+    thrown.expect(IllegalStateException.class);
```

Comment:
Suggest checking the exception message as well, or at least enough to verify it is the message you expected (eg., "after finishBundle()"). The reason is that we tend to add exceptions over time. Often, we'll have a test like this, then something else will start throwing an IllegalStateException (for some new reason). Then we'll no longer be testing what we thought we were, but we won't know it. Then we'll break it.